public/app/features/alerting/unified/components/permissions/ManagePermissions.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState, ComponentProps } from 'react';
+
+import { Button, Drawer } from '@grafana/ui';
+import { Permissions } from 'app/core/components/AccessControl';
+import { t, Trans } from 'app/core/internationalization';
+
+type ButtonProps = { onClick: () => void };
+
+type BaseProps = Pick<ComponentProps<typeof Permissions>, 'resource' | 'resourceId'> & {
+  resourceName?: string;
+  title?: string;
+};
+
+type Props = BaseProps & {
+  renderButton?: (props: ButtonProps) => JSX.Element;
+};
+
+/**
+ * Renders just the drawer containing permissions management for the resource.
+ *
+ * Useful for manually controlling the state/display of the drawer when you need to render the
+ * controlling button within a dropdown etc.
+ */
+export const ManagePermissionsDrawer = ({
+  resourceName,
+  title,
+  onClose,
+  ...permissionsProps
+}: BaseProps & Pick<ComponentProps<typeof Drawer>, 'onClose'>) => {
+  const defaultTitle = t('alerting.manage-permissions.title', 'Manage permissions');
+  return (
+    <Drawer onClose={onClose} title={title || defaultTitle} subtitle={resourceName}>
+      <Permissions {...permissionsProps} canSetPermissions></Permissions>
+    </Drawer>
+  );
+};
+
+/** Default way to render the button for "manage permissions" */
+const DefaultButton = ({ onClick }: ButtonProps) => {
+  return (
+    <Button variant="secondary" onClick={onClick} icon="unlock">
+      <Trans i18nKey="alerting.manage-permissions.button">Manage permissions</Trans>
+    </Button>
+  );
+};
+
+/**
+ * Renders a button that opens a drawer with the permissions editor.
+ *
+ * Provides capability to render button as custom component, and manages open/close state internally
+ */
+const ManagePermissions = ({ resource, resourceId, resourceName, title, renderButton }: Props) => {
+  const [showDrawer, setShowDrawer] = useState(false);
+  const closeDrawer = () => setShowDrawer(false);
+  const openDrawer = () => setShowDrawer(true);
+
+  return (
+    <>
+      {renderButton ? renderButton({ onClick: openDrawer }) : <DefaultButton onClick={openDrawer} />}
+      {showDrawer && (
+        <ManagePermissionsDrawer
+          resource={resource}
+          resourceId={resourceId}
+          resourceName={resourceName}
+          title={title}
+          onClose={closeDrawer}
+        />
+      )}
+    </>
+  );
+};
+
+export default ManagePermissions;
```

Comment:
How about switching to a named export? Default exports can be harder for IDEs to import, tricky to navigate, and a bit confusing if renamed in the import. Named exports are usually clearer and easier to work with

Reply:
Sure, I can change it :+1: I think that we should work out some style/linting rules to apply (at least to the alerting code) so we're consistent about this kind of thing across the codebase.